src/Microsoft.AspNetCore.Mvc.Core/Internal/MvcCoreLoggerExtensions.cs

```diff
@@ -639,7 +645,22 @@ public static IDisposable ActionScope(this ILogger logger, ActionDescriptor acti
 
         public static void ExecutingAction(this ILogger logger, ActionDescriptor action)
         {
-            _actionExecuting(logger, action.DisplayName, null);
+            var routeKeys = action.RouteValues.Keys.ToArray();
+            var routeValues = action.RouteValues.Values.ToArray();
+            string route = "\"";
+            for (var i = 0; i < routeValues.Count(); i++)
+            {
+                if (i == routeValues.Count() - 1)
```

Comment:
Avoid using `.Count()` - it's almost always a sign that there's a more efficient way. In you case you already called `.ToArray()` so you can use `.Length` which is muuuuuuuuuuuuuuuuuch more efficient.